### PR TITLE
resourcequota return StatusError when timeout

### LINKED ISSUE #64244: ErrorToAPIStatus could be more informative about unexpected errors
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind feature

**What happened**:
I am suffering from mysterious timeouts.  See https://github.com/kubernetes/kubernetes/issues/63608#issuecomment-391497458 for the evidence from my latest investigation.  It includes this line

```
E0523 18:59:53.086835       1 status.go:64] apiserver received an error that is not an metav1.Status: timeout
```

from the `ErrorToAPIStatus` function in `staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/status.go`.   The relevant line is this:

```
	runtime.HandleError(fmt.Errorf("apiserver received an error that is not an metav1.Status: %v", err))
```

I still do not know the type of the error in question or where it is coming from.  It would be helpful if the log message included the type of the error object --- such as with `fmt.Sprintf("%T", err)` or, if that does not include the type's package name, perhaps something that would really nail the type.

**What you expected to happen**:
I would expect more information about an unexpected type of error.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.10.3
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Fixes #64244

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/admission/resourcequota/controller.go [hunk 1]
```diff
@@ -549,7 +550,7 @@ func (e *quotaEvaluator) Evaluate(a admission.Attributes) error {
 	select {
 	case <-waiter.finished:
 	case <-time.After(10 * time.Second):
-		return fmt.Errorf("timeout")
+		return apierrors.NewInternalError(fmt.Errorf("resource quota evaluates timeout"))
```
Comment (lavalamp) [inline plugin/pkg/admission/resourcequota/controller.go]: I don't understand this error message. Is it trying to say `resource quota evaluation took too long`?

Comment (hzxuzhonghu) [reply plugin/pkg/admission/resourcequota/controller.go]: yes

Comment (MikeSpreitzer) [reply plugin/pkg/admission/resourcequota/controller.go]: I like "resource quota evaluation took too long" better.

**Review by MikeSpreitzer — COMMENTED** [review]
While I appreciate an improvement here, it does not really resolve issue #64244 --- which is about how errors of unexpected type are handled.

**Pull request merged**